### ISSUE TITLE
Align server timeout with ES_TIMEOUT

### ISSUE THREAD

**Issue by connectwithnara** [openzipkin/zipkin#2809]
The server times out before the ES client times out, when the ES_TIMEOUT is set to a high value like 60s.

Align the server timeout to match with the ES_TIMEOUT value either automatically or add necessary documentation to avoid such conflicts.

**Comment by codefromthecrypt** [comment]
@anuraaga feel free to kick back if no room on plate

**Comment by anuraaga** [comment]
@narayaruna Want to confirm one thing - are you seeing issues with span collection or with the zipkin server health endpoint? Span collection only parses spans before defering write to storage and immediately returning a response so a short server timeout should be ok. But the health endpoint doesn't have this property so is it where you're seeing the server timeout?

**Comment by codefromthecrypt** [comment]
This was a read query, not a storage request. I asked Nara to bump from a
chat he mentioned it in.

**Comment by codefromthecrypt** [comment]
`ZipkinServerConfiguration` has a magic number:

```
      // It's common for backend requests to have timeouts of the magic number 10s, so we go ahead
      // and default to a slightly longer timeout on the server to be able to handle these with
      // better error messages where possible.
      sb.requestTimeout(Duration.ofSeconds(11));
```

**Comment by codefromthecrypt** [comment]
@narayaruna how about we expose QUERY_TIMEOUT for magic 11 seconds. Then, it can be updated manually. This follows prefix of other query related parameters.

**Comment by anuraaga** [comment]
Going to bounce this back for now can get back to it next week if still there, don't think I'll be at a computer.

**Comment by anuraaga** [comment]
This knob was released in 2.19.2
